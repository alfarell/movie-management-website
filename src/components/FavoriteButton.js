import React, { useState } from 'react';
import { Button } from 'antd';
import { HeartFilled } from '@ant-design/icons';

const FavoriteButton = ({ label, type, onClick, style, iconStyle }) => {
    const [showLabel, setShowLabel] = useState(false);

    return (
        <Button
            danger
            id='favorite-button'
            type={type}
            icon={<HeartFilled style={iconStyle} />}
            style={style}
            onMouseEnter={() => setShowLabel(true)}
            onMouseLeave={() => setShowLabel(false)}
            onClick={onClick}
        >
            {showLabel ? label : null}
        </Button>
    );
};

export default FavoriteButton;
