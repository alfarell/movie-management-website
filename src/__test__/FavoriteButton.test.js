import React from 'react'
import { render, act, fireEvent } from '@testing-library/react';
import FavoriteButton from '../components/ButtonComponent/FavoriteButton';


describe('Favorite Button', () => {
    it('render favorite button', () => {
        const component = render(
            <FavoriteButton label={'Add to Favorite'} hideMode={false} />
        );

        expect(component).toMatchSnapshot();
        expect(component.getByTestId('favorite-button-test')).toHaveTextContent('Add to Favorite');
    });

    it('render favorite button hide mode(hidden label)', () => {
        const component = render(
            <FavoriteButton label={'Add to Favorite'} hideMode={true} />
        );

        expect(component).toMatchSnapshot();
        expect(component.getByTestId('favorite-button-test')).not.toHaveTextContent('Add to Favorite');
    });

    describe('Hover on Favorite Button hidden mode/hidden label', () => {
        it('should show label on mouse enter', () => {
            const { getByTestId, getByText } = render(
                <FavoriteButton label={'Add to Favorite'} hideMode={true} />
            );

            act(() => {
                fireEvent.mouseEnter(getByTestId('favorite-button-test'));
            });

            expect(getByText('Add to Favorite')).toBeInTheDocument();
            expect(getByTestId('favorite-button-test')).toHaveTextContent('Add to Favorite');
        });

        it('should hide label on mouse leave', () => {
            const { getByTestId, getByText } = render(
                <FavoriteButton label={'Add to Favorite'} hideMode={true} />
            );

            act(() => {
                fireEvent.mouseEnter(getByTestId('favorite-button-test'));
            });

            expect(getByText('Add to Favorite')).toBeInTheDocument();
            expect(getByTestId('favorite-button-test')).toHaveTextContent('Add to Favorite');

            act(() => {
                fireEvent.mouseLeave(getByTestId('favorite-button-test'));
            });

            expect(getByTestId('favorite-button-test')).not.toHaveTextContent('Add to Favorite');
        });
    });

});