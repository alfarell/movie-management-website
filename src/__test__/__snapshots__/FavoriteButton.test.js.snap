// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite Button render favorite button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="ant-btn ant-btn-dangerous"
        data-testid="favorite-button-test"
        style="z-index: 1;"
        type="button"
      >
        <span
          aria-label="heart"
          class="anticon anticon-heart"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="heart"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M923 283.6a260.04 260.04 0 00-56.9-82.8 264.4 264.4 0 00-84-55.5A265.34 265.34 0 00679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 00-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9z"
            />
          </svg>
        </span>
        <span>
          Add to Favorite
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="ant-btn ant-btn-dangerous"
      data-testid="favorite-button-test"
      style="z-index: 1;"
      type="button"
    >
      <span
        aria-label="heart"
        class="anticon anticon-heart"
        role="img"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="heart"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M923 283.6a260.04 260.04 0 00-56.9-82.8 264.4 264.4 0 00-84-55.5A265.34 265.34 0 00679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 00-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9z"
          />
        </svg>
      </span>
      <span>
        Add to Favorite
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Favorite Button render favorite button hide mode(hidden label) 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="ant-btn ant-btn-icon-only ant-btn-dangerous"
        data-testid="favorite-button-test"
        style="z-index: 1;"
        type="button"
      >
        <span
          aria-label="heart"
          class="anticon anticon-heart"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="heart"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M923 283.6a260.04 260.04 0 00-56.9-82.8 264.4 264.4 0 00-84-55.5A265.34 265.34 0 00679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 00-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9z"
            />
          </svg>
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="ant-btn ant-btn-icon-only ant-btn-dangerous"
      data-testid="favorite-button-test"
      style="z-index: 1;"
      type="button"
    >
      <span
        aria-label="heart"
        class="anticon anticon-heart"
        role="img"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="heart"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M923 283.6a260.04 260.04 0 00-56.9-82.8 264.4 264.4 0 00-84-55.5A265.34 265.34 0 00679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 00-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9z"
          />
        </svg>
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
